src/main/java/org/fcrepo/camel/FcrepoProducer.java

```diff
@@ -99,10 +100,10 @@ protected void doInTransactionWithoutResult(final TransactionStatus status) {
     private void doRequest(final Exchange exchange, final String transaction) throws FcrepoOperationFailedException {
         final Message in = exchange.getIn();
         final HttpMethods method = getMethod(exchange);
-        final String contentType = getContentType(exchange);
-        final String accept = getAccept(exchange);
         final String url = getUrl(exchange, transaction);
-        final String prefer = getPrefer(exchange);
+        final String contentType = Optional.ofNullable(endpoint.getContentType())
+                                           .filter(x -> x.length() > 0)
```

Comment:
Actually, maybe better [StringUtils::isNoneEmpty](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNoneEmpty%28java.lang.CharSequence...%29) to avoid the negation.